shell/browser/api/electron_api_tray.cc

```diff
@@ -236,14 +262,20 @@ void Tray::DisplayBalloon(gin_helper::ErrorThrower thrower,
 }
 
 void Tray::RemoveBalloon() {
+  if (!CheckDestroyed())
+    return;
   tray_icon_->RemoveBalloon();
 }
 
 void Tray::Focus() {
+  if (!CheckDestroyed())
+    return;
   tray_icon_->Focus();
 }
 
-void Tray::PopUpContextMenu(gin_helper::Arguments* args) {
+void Tray::PopUpContextMenu(gin::Arguments* args) {
+  if (!CheckDestroyed())
+    return;
   gin::Handle<Menu> menu;
   args->GetNext(&menu);
```

Comment:
The `gin::Arguments` would increase `nextArg` counter even when `GetNext` fails, so `tray.popUpContextMenu(pos)` would not work with current parsing code.

Reply:
Ah, good catch. I've cleaned up the argument parsing logic and added some tests. The new code should be better in that it will properly throw errors for invalid argument types, rather than silently using the default behavior.